File: app/src/main/java/vn/loitp/app/activity/pattern/observerpattern/Subject.java
package vn.loitp.app.activity.pattern.observerpattern;

/**
 * Created by www.muathu@gmail.com on 1/31/2018.
 */

public interface Subject {
    void registerObserver(RepositoryObserver repositoryObserver);

    void removeObserver(RepositoryObserver repositoryObserver);

    void notifyObservers();
}

File: app/src/main/java/vn/loitp/app/activity/pattern/observerpattern/Subject.kt
package vn.loitp.app.activity.pattern.observerpattern

/**
 * Created by www.muathu@gmail.com on 1/31/2018.
 */

interface Subject {
    fun registerObserver(repositoryObserver: RepositoryObserver)

    fun removeObserver(repositoryObserver: RepositoryObserver)

    fun notifyObservers()
}
